apps/examples/testcase/le_tc/pm/tc_pm_main.c

```diff
@@ -0,0 +1,214 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <tinyara/pm/pm.h>
+#include <tinyara/timer.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <tc_common.h>
+
+#define PM_DRVPATH           "/dev/pm"
+#define SUSPEND_TIME         5000           // It is in millisecond
+#define SLEEP_TIME           5000           // It is in millisecond
+#define TOTAL_CHECK_TIME     20000          // It is in millisecond
+#define MISSING_TICK         5000 
+#define DEVIATION_TICK       200
+
+static int pm_fd;
+
+int get_pm_fd(void)
+{
+    	return pm_fd;
+}
+
+/****************************************************************************
+ * Name: tc_pm_timedstay
+ *
+ * Description:
+ *   Testing of pm timedstay operation. This function runs timedstay function
+ *   and checks if the pm state's stay count increases during timer running and
+ *   goes back to initial value after timer is expired. 
+ *
+ * Returned Value:
+ *   None
+ ****************************************************************************/
+void tc_pm_timedstay(void)
+{
+    	int ret_chk;
+    	int curr_timetick;
+    	int last_timetick;
+    	int initial_staycount, during_staycount, final_staycount;
+
+    	/* Storing the stay count before timedstay */
+    	initial_staycount = ioctl(pm_fd, PMIOC_STAYCOUNT, PM_NORMAL);
+    	TC_ASSERT_GEQ("pm_ioctl staycount", initial_staycount, OK);
+
+    	int time_fd = open("/dev/timer0", O_WRONLY);
+    	TC_ASSERT_GEQ("time_ioctl gettime", time_fd, OK);
+
+    	curr_timetick = ioctl(time_fd, TCIOC_GETTIMETICK, NULL);
+    	last_timetick = curr_timetick;
+
+    	pm_suspend_arg_t arg = {PM_NORMAL, SUSPEND_TIME};
+    	ret_chk = ioctl(pm_fd, PMIOC_TIMEDSUSPEND, &arg);
+    	TC_ASSERT_EQ("pm_ioctl timedstay", ret_chk, OK);
+
```

Comment:
For the correct test, pm_relax must be called once here.
Because we are holding pm_stay while booting.